### PR TITLE
mididings package

### PR DESCRIPTION
**Pull request by whohoho**
###### Motivation for this change

New package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/mididings/0001-add-boost-suffix.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
++++ b/setup.py
+@@ -128,7 +128,7 @@ def boost_lib_name(name, add_suffixes=[]):
+     Try to figure out the correct boost library name (with or without "-mt"
+     suffix, or with any of the given additional suffixes).
+     """
+-    for suffix in add_suffixes + ['', '-mt']:
++    for suffix in add_suffixes + ['', '-mt', '36']:
+         for libdir in lib_dirs():
+             for ext in ['so'] + ['dylib'] * (sys.platform == 'darwin'):
+                 libname = 'lib%s%s.%s' % (name, suffix, ext)
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/0001-add-boost-suffix.patch]: It is interesting that there is no symlink without the version suffix.

File: pkgs/development/python-modules/mididings/default.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  buildInputs = with pkgs; [
+      (boost.override {
+        enablePython = true;
+        inherit python;
+      })
+  	 glib
+         libjack2
+	 alsaLib
+    	pkgconfig
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: Not needed here.

File: pkgs/development/python-modules/mididings/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      })
+  	 glib
+         libjack2
+	 alsaLib
+    	pkgconfig 
+    ];
+
+  runtimeDeps = [
+    python36Packages.decorator
+  ];
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: This should be removed.

File: pkgs/development/python-modules/mididings/default.nix [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                   python36Packages.pyinotify 
+                   python36Packages.tkinter  ];
+
+
+   checkInputs = [ python36Packages.pytest 
+ 		   python36Packages.decorator 
+		   python36Packages.pyliblo 
+                   python36Packages.dbus-python
+                   python36Packages.pyinotify 
+                   python36Packages.tkinter  ];
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: You only need to add additional packages here. No need to duplicate `propagatedBuildInputs`.
Pytest should be here but not in `propagatedBuildInputs`.

File: pkgs/development/python-modules/mididings/default.nix [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  src = fetchFromGitHub {
+    owner = "dsacre";
+    repo = "mididings";
+    rev = "bbec99a8c878a2a7029e78e84fc736e4a68ed5a0";
+    sha256 = "1pdf5mib87zy7yjh9vpasja419h28wvgq6x5hw2hkm7bg9ds4p2m";
+  };
+  #doCheck = false;
+  #checkPhase = ''
+  #  echo check
+  #'';
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: Please remove

File: pkgs/development/python-modules/mididings/default.nix [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "mididings";
+  version = "unstable-2015-11-17";
+
+
+  patches = [ ./0001-add-boost-suffix.patch ];
+
+  LIBRARY_PATH="${(pkgs.boost.override {enablePython = true; inherit python; })}/lib";
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: What is the purpose of this?

File: pkgs/development/python-modules/mididings/default.nix [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  };
+  #doCheck = false;
+  #checkPhase = ''
+  #  echo check
+  #'';
+
+  meta = {
+    description = "midi router";
+    homepage = https://github.com/dsacre/mididings;
+    license = lib.licenses.gpl3;
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: Are you going to maintain this package?

File: pkgs/development/python-modules/mididings/default.nix [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+, pkgs
+, libjack2
+, alsaLib
+, pkgconfig
+, glib
+, boost
+, python36Packages
+}:
+
+buildPythonPackage rec {
```
Comment (Mic92) [inline pkgs/development/python-modules/mididings/default.nix]: As this is an application please use `buildPythonApplication` and move it out of `python-modules` into a different directory.

**Review by Mic92 — CHANGES_REQUESTED** [review]
The project looks a bit unmaintained. Is this still actively developed? Please fix the indentation in the file. We use two spaces for indentation.

File: pkgs/development/python-modules/mididings/default.nix (hunk 6 above)
Comment (whohoho) [reply pkgs/development/python-modules/mididings/default.nix]: Yes, without it it gives a linker error: 
/nix/store/vcc4svb8gy29g4pam2zja6llkbcwsyiq-binutils-2.30/bin/ld: cannot find -lboost_python

File: pkgs/development/python-modules/mididings/default.nix (hunk 8 above)
Comment (whohoho) [reply pkgs/development/python-modules/mididings/default.nix]: If it is done as a python application, can you still import it as a library?, cause that is how it is mostly used.

File: pkgs/development/python-modules/mididings/default.nix (hunk 7 above)
Comment (whohoho) [reply pkgs/development/python-modules/mididings/default.nix]: i put myself as the maintainer

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by mmahut** [comment]
Are there any updates on this pull request, please?

**Comment by gnidorah** [comment]
Closed by #63207

**Pull request closed without merging**